lib/board.rb

```diff
@@ -56,10 +56,9 @@ def consecutive?(coordinates)
 
   def not_diagonal?(coordinates)
     row_letter = coordinates.first[0]
-    column_number = coordinates.first[1]
-    coordinates.all? do |coordinate|
-      coordinate[0] == row_letter || coordinate[1] == column_number
-    end
+    column_number = coordinates.first[1..2]
+    coordinates.all? {|coordinate| coordinate[0] == row_letter} ||
+    coordinates.all? {|coordinate| coordinate[1..2] == column_number}
```

Comment:
I noticed that some invalid placements were seeping into our valid_placements array in the Battle class. Revisited not_diagonal? to beef it up and fix the issue.

Original code: Was one test, which returned `true` if all the coordinates had either the same letter OR number as first coordinate. The set `["A1", "A2", "C1"]` was passing, because `A1` and `A2` have the same letter, and `A1` and `C1` have the same number. Since it was letter OR number could match the first coordinate, it slipped under the radar as "valid".

New code: runs TWO tests. First test: if all coordinates have the same letter. Second test: if all coordinates have the same number. As long as one of those checks runs `true` then the coordinates are valid.